### PR TITLE
Partial revert of #10433 (Texture Cache Util: Fix block depth adjustment on slices)

### LINKED ISSUE #11252: Asterix and Obelix XXL graphical issues
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

mainline 1518 

### Description of Issue

The game suffers from texture corruption and missing particles on this emulator:
image

### Expected Behavior

Game is supposed to look like this:
image

### Reproduction Steps

Just open the game on the emulator and get trought the menus

### Log File

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/12307803/yuzu_log.txt)

### System Configuration

CPU: r 5600g
GPU/Driver: gtx 1050 ti (536.67) 
RAM: 16g 3200mz
OS: Windows 11

### PR DESCRIPTION
**Pull request by Squall-Leonhart**
Alternative to #11528 

The If block in this change was causing some 2D textures to be treated as if their mip 0 was a 3D Slice, this could be ascertained as the same texture viewed from different distances would render fine, but then close up would look like a decoding failure.

It also resulted in some 3D ASTC textures not being scaled appropriate leading to broken graphical effects such as the jagged TOTK recall animation being a circle, as the If block was only accepting the image based on its original info without any adjustments applied.

Fixes #11252  and #10493

Doesn't reintroduce #10255

### CODE REVIEW

**Review by FernandoS27 — APPROVED** [review]

File: src/video_core/texture_cache/util.cpp [hunk 1]
```diff
@@ -1294,9 +1287,9 @@ u32 MapSizeBytes(const ImageBase& image) {
 
 static_assert(CalculateLevelSize(LevelInfo{{1920, 1080, 1}, {0, 2, 0}, {1, 1}, 2, 0}, 0) ==
               0x7f8000);
-static_assert(CalculateLevelSize(LevelInfo{{32, 32, 1}, {0, 0, 4}, {1, 1}, 4, 0}, 0) == 0x40000);
+static_assert(CalculateLevelSize(LevelInfo{{32, 32, 1}, {0, 0, 4}, {1, 1}, 4, 0}, 0) == 0x4000);
```
Comment (liamwhite) [inline src/video_core/texture_cache/util.cpp:1290]: The level 0 size of a 32x32x1 image with block dim {0,0,4}, tile size {1,1}, 16bpp, and 0 tile spacing is in fact 0x40000.

First consider the case with block dim {0,0,0}:
- At 16bpp, with 64x8 gob size and tile size {1,1}, each gob must cover a 4x8 area.
- For 32x32x1, there must be 8 gobs in x, 4 gobs in y, 1 gob in z
- This gives a level size of 8\*4\*1\*gobSize, or 0x4000

Then consider the case with block dim {0,0,4}.
The same considerations apply, except we must use the value given by the block dimensions instead, as this is a slice in z:
- At 16bpp, with 64x8 gob size and tile size {1,1}, each gob must cover a 4x8 area.
- For 32x32x1, there must be 8 gobs in x and 4 gobs in y, and we must use the z value which is 2^4=16 gobs in z
- This gives a level size of 8\*4\*16\*gobSize, or 0x40000

So this test case or code is incorrect in some way, even if all observable behavior is okay.

**Review by liamwhite — DISMISSED** [review]

File: src/video_core/texture_cache/util.cpp (hunk 1 above)
Comment (Squall-Leonhart) [reply src/video_core/texture_cache/util.cpp:1290]: Not sure tbh, this is the original static assert, any other value is a test failure, including 0x40000.

**Comment by Aphcity** [comment]
I checked the package generated by Actions on windows. And the translation pack for Tsukihime could still cause the same problem mentioned above. https://github.com/yuzu-emu/yuzu/issues/10456

image

**Comment by Squall-Leonhart** [comment]
> I checked the package generated by Actions both on windows and Android. And the translation pack for Tsukihime could still cause the same problem mentioned above. #10456
> 
> image

So its the AdjustMipBlock change breaking that title, I have an idea....., try the next Action

**Comment by gidoly** [comment]
ready for ea test couldn't find hiccup on games i tested

**Comment by liamwhite** [comment]
Well, I can't explain why since the computed sizes don't match my expectation, but both gloom and recall animation in TOTK seem to now work just fine on my extremely memory-limited 560.
image

**Comment by Squall-Leonhart** [comment]
> ready for ea test couldn't find hiccup on games i tested

I want to hear from the Tsukuhime owner first, and Liam really wants the asserts set correctly.

**Comment by Aphcity** [comment]
Having run another two artifacts with Tsukihime on Windows, the issue is still.

The artifacts are generated by https://github.com/yuzu-emu/yuzu/commit/02b897ce27f08c0b120db207ea47edd0cfa7998c and https://github.com/yuzu-emu/yuzu/commit/238e9affe6983834ed1446a7d000d5b6b5669a8d

image

image

**Comment by Squall-Leonhart** [comment]
is this the latest pack for tsukuhime? the fix they made for it to work with the prior changes might not work with these ones.

**Comment by Aphcity** [comment]
> is this the latest pack for tsukuhime? the fix they made for it to work with the prior changes might not work with these ones.

No, the trans pack was an old one made in Sep 2022.

**Comment by Squall-Leonhart** [comment]
I need one of the affected textures to look at its structure.

> > is this the latest pack for tsukuhime? the fix they made for it to work with the prior changes might not work with these ones.
> 
> No, the trans pack was an old one made in Sep 2022.

Can you check with this

[Release.zip](https://github.com/yuzu-emu/yuzu/files/12667648/Release.zip)

**Comment by Aphcity** [comment]
> I need one of the affected textures to look at its structure.
> 
> > > is this the latest pack for tsukuhime? the fix they made for it to work with the prior changes might not work with these ones.
> > 
> > 
> > No, the trans pack was an old one made in Sep 2022.
> 
> Can you check with this
> 
> [Release.zip](https://github.com/yuzu-emu/yuzu/files/12667648/Release.zip)

Yes! It works completely right now.

**Comment by Squall-Leonhart** [comment]
> > I need one of the affected textures to look at its structure.
> > > > is this the latest pack for tsukuhime? the fix they made for it to work with the prior changes might not work with these ones.
> > > 
> > > 
> > > No, the trans pack was an old one made in Sep 2022.
> > 
> > 
> > Can you check with this
> > [Release.zip](https://github.com/yuzu-emu/yuzu/files/12667648/Release.zip)
> 
> Yes! It works completely right now.

investigating this right now to identify exactly what is up with Layout 64 and how it differs to 0, my check of course didn't work because these are single mip textures so they are always going the route that isn't playing nice with them.

**Comment by Aphcity** [comment]
> > > I need one of the affected textures to look at its structure.
> > > > > is this the latest pack for tsukuhime? the fix they made for it to work with the prior changes might not work with these ones.
> > > > 
> > > > 
> > > > No, the trans pack was an old one made in Sep 2022.
> > > 
> > > 
> > > Can you check with this
> > > [Release.zip](https://github.com/yuzu-emu/yuzu/files/12667648/Release.zip)
> > 
> > 
> > Yes! It works completely right now.
> 
> investigating this right now to identify exactly what is up with Layout 64 and how it differs to 0, my check of course didn't work because these are single mip textures so they are always going the route that isn't playing nice with them.

you mean that we can't fix it with other issues like https://github.com/yuzu-emu/yuzu/issues/11252 now caused by Layout 0. 

Maybe you can send me another Android APK package to test if it works, and you can call me after any commit to test if it fixes tsukihime.

**Comment by Squall-Leonhart** [comment]
It looks like the log2 byte size is too small, and changing this breaks all the assert tests, its set to 0x1F and atleast 0x20 is needed for Format 64 to work.

**Comment by Aphcity** [comment]
> I need one of the affected textures to look at its structure.
> 
> > > is this the latest pack for tsukuhime? the fix they made for it to work with the prior changes might not work with these ones.
> > 
> > 
> > No, the trans pack was an old one made in Sep 2022.
> 
> Can you check with this
> 
> [Release.zip](https://github.com/yuzu-emu/yuzu/files/12667648/Release.zip)
> 
> My check doesn't work because this is a single mip BNTX format, so of course its passing into block_size.depth and not AdjustMipBlockSize

I tried to remove in my personal fork but failed. May I have your commit source code on this version?

**Comment by boludoz** [comment]
Someone can check that? #11600 I think it also has something to do with it.

**Comment by liamwhite** [comment]
there is no evidence that 11600 has anything to do with this, please keep discussions on topic

**Comment by Squall-Leonhart** [comment]
Partially Fixes #11666, for some other glitches copy from D32_FLOAT to A8B8G8R8_UNORM is required.

**Comment by gidoly** [comment]
break the legend of zelda tears of the kingdom grass rendering after 5 minutes of game play
yuzu-msvc-verify-pr11534-20230919-fe7710a76_5vaPtC1LCG

**Comment by Squall-Leonhart** [comment]
this change doesn't do anything to grass

> break the legend of zelda tears of the kingdom grass rendering after 1~5 minutes of game play yuzu-msvc-verify-pr11534-20230919-fe7710a76_5vaPtC1LCG

You sure it was this change?

**Comment by gidoly** [comment]
yeah i checked master

**Comment by Squall-Leonhart** [comment]
> yeah i checked master and pulled branch manually grass disappears in 1~5 minof gameplay

this PR wasn't applied to master yet.

**Comment by gidoly** [comment]
eeer sorry seems i have disabled reactive flushing accidently. sorry for wasting your time.

**Comment by igoticecream** [comment]
Is the recall animation supposed to look "too simetrical"? Attaching a screenshot of 3911 and real hardware

yuzu 2023-10-07 11-47-47
image

**Comment by Squall-Leonhart** [comment]
I've tested up to 3x Yuzu scaling and it appears like the bottom of the two images for me,  are you perhaps using Chucks 1.5.5b3 mod beyond 1440p?

**Comment by igoticecream** [comment]
ah yes, seems like chuck's fps/resolution mod is messing with the animation,  without it i see the correct animation now

**Pull request merged**